### ISSUE TITLE
NPCs no longer plant tiles

### ISSUE THREAD

**Issue by Funguss** [CleverRaven/Cataclysm-DDA#38565]
Got a killer headache so pushing this a bit. Not fully investigated, but I guess it's due to #38480, could you have a look over again @jkraybill ? Got six NPCs, five set to farm, they all just stand around staring at the ground and occasionally sleep and eat. Save: [Shadeland.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4289866/Shadeland.zip)

# Steps To Reproduce

1. Just farm with NPCs I think, or use multiple NPCs.

# Versions and configuration

- OS: Windows
    - OS Version: MINGW/CYGWIN/MSYS2 on unknown Windows version
- Game Version: 0.D-12851-g616e03e [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Makeshift Items Mod [makeshift],
    More Survival Tools [More_Survival_Tools],
    Mythological Replicas [nw_pack],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    No Rail Stations [No_Rail_Stations],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Roadheader and other mining vehicles [Heavy miners],
    No Fungal Monsters [No_Fungi],
    Beta National Guard Camp [national_guard_camp]
]

**Comment by jkraybill** [comment]
Yes will look at this!

**Comment by Funguss** [comment]
Thanks, sorry if that came across as rude, still got a terrific headache. Thought as you did the PR giving you a shout was a good call? Looking over the PR properly it looks like you didn't get the code review you wanted! Grahh!

Also, regarding camps, it's been a while but last time I had one all camp stuff was abstracted away, the NPC disappeared then reappeared when you interacted with the job via the board. I think davidpwbrown changed that in a PR but it's in feature freeze. Depending on which activity you used for reference you may have overlooked some specifics for checking who is performing the activity, that's my best guess. Using #38505 as a reference may be good, I'm not sure and at my limit.

... I type all that and still sit here staring and thinking. The only insight I've gained is that you might find #37300 useful if you pull it to your fork.

**Comment by jkraybill** [comment]
Not at all - very helpful and I agree that my commit is a possibly cause. But per my notes on that PR -- prior to my changes I couldn't actually get faction farming to work, similar to what you saw. Do you know approximately the most recent version you saw this work, and can you also let me know the steps to get faction farming working? I couldn't get it to work but it was my first time playing with a proper faction camp as I'm a lone wolfer.

**Comment by Funguss** [comment]
Urgh, it would have been months ago...

Making use of my insomnia, I also couldn't get faction camps to work. I wasn't talking about faction camps specifically in my bug report, or at least didn't mean to; I'm pretty certain that none of the camp tasks start activities, they're just simulated... Except maybe woodcutting, though if it's off screen it is. I did manage to get the farm plot option to appear but it greyed out once I spawned in some wheat seeds. This seems like an unrelated bug.

The issue I meant was setting NPCs on your farm plots.

1. Set up farm plots.
2. Give an NPC a digging tool and some seeds or drop them in a seed pile.
3. Ask them to farm (should be 'i' then 'l').
4. Watch them dig about a bit then stand there.
